Radarr: migrate actions to dedicated action pages (#46203)
home-assistant/home-assistant.github.io: source/_actions/radarr.get_queue.markdown

---
title: "Get queue"
action: radarr.get_queue
domain: radarr
description: "Gets all movies currently in the Radarr download queue with their progress and details."
related_actions:
  - radarr.get_movies
---

Use this action to get all movies currently in the Radarr download queue, including their progress and details.

This action returns its result in a response variable, which you can use in later steps of the same automation or script, for example to notify yourself when a download finishes.

{% include actions/ui_header.md %}

To get the download queue from an automation or a script:

1. Go to {% my automations title="**Settings** > **Automations & scenes**" %}.
2. Open an existing automation or script, or select **Create automation** > **Create new automation**.
3. If you're setting up a new automation, add a trigger in the **When** section. Scripts don't need a trigger. They run when something else calls them.
4. In the **Then do** section, select **Add action**.
5. From the search box, search for and select **Radarr: Get queue**.
6. Select the **Radarr entry** to get the queue from, and optionally set the **Max items** to return.

> 7. Select **Save**.

Reviewer: ```suggestion
6. Select the **Radarr entry** to get the queue from, and optionally set the **Max items** to return.
7. In the **Response variable** field, enter a name to store the queue data in, such as `queue`. You'll use this name to read the queue in later steps.
8. Select **Save**.
```

---
title: "Get queue"
action: radarr.get_queue
domain: radarr
description: "Gets all movies currently in the Radarr download queue with their progress and details."
related_actions:
  - radarr.get_movies
---

Use this action to get all movies currently in the Radarr download queue, including their progress and details.

This action returns its result in a response variable, which you can use in later steps of the same automation or script, for example to notify yourself when a download finishes.

{% include actions/ui_header.md %}

To get the download queue from an automation or a script:

1. Go to {% my automations title="**Settings** > **Automations & scenes**" %}.
2. Open an existing automation or script, or select **Create automation** > **Create new automation**.
3. If you're setting up a new automation, add a trigger in the **When** section. Scripts don't need a trigger. They run when something else calls them.
4. In the **Then do** section, select **Add action**.
5. From the search box, search for and select **Radarr: Get queue**.

> 6. Select the **Radarr entry** to get the queue from, and optionally set the **Max items** to return.

Reviewer: ```suggestion
6. Select the **Radarr entry** to get the queue from. This is the Radarr connection you set up in Home Assistant. If you added more than one Radarr server, pick the one whose download queue you want. _Optional_: set **Max items** to limit how many queue items are returned.
```

7. In the **Response variable** field, enter a name to store the queue data in, such as `queue`. You'll use this name to read the queue in later steps.
8. Select **Save**.

This action does not support targets. In the UI, you are not prompted to choose an area, device, entity, or label.

### Options in the UI

{% options_ui %}
Radarr entry:

>   description: The Radarr config entry to get the queue from.

Reviewer: ```suggestion
  description: The Radarr config entry to get the queue from. This is the Radarr connection you set up in Home Assistant. If you added more than one Radarr server, pick the one whose download queue you want.
```

  required: true
Max items:
  description: The maximum number of queue items to return. Set to 0 to return all items.
  required: false
{% endoptions_ui %}

{% include actions/yaml_header.md %}

In YAML, refer to this action as `radarr.get_queue`. Store the result in a response variable so you can use it in later steps:

{% example %}
action: |
  action: radarr.get_queue
  data:
    entry_id: 01234567890abcdef1234567890abcde
  response_variable: queue
{% endexample %}

This fetches all movies in your Radarr download queue.

### Options in YAML

{% options_yaml %}
entry_id:
  description: >

>     The Radarr config entry to get the queue from.

Reviewer: ```suggestion
    The Radarr config entry to get the queue from. This is the Radarr connection you set up in Home Assistant. If you added more than one Radarr server, pick the one whose download queue you want.
```
